src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -376,32 +367,10 @@ internal static bool IsValidHostObjectType(Type type)
         /// <paramref name="hasValue"/> is false in the former case and true
         /// in the latter.
         /// </remarks>
-        public ITypeSymbol GetSubmissionResultType(out bool hasValue)
-        {
-            return CommonGetSubmissionResultType(out hasValue);
-        }
-
-        protected abstract ITypeSymbol CommonGetSubmissionResultType(out bool hasValue);
+        internal abstract ITypeSymbol GetSubmissionResultType(out bool hasValue);
```

Comment:
These are independent values though. hasValue =false doesn't mean the return value is null. In any case, I'm probably gonna replace this logic with something else at some point.